src/promptflow/promptflow/_sdk/entities/_flow.py

```diff
@@ -341,12 +341,12 @@ def __call__(self, *args, **kwargs):
 
     def invoke(self, inputs: dict) -> "LineResult":
         """Invoke a flow and get a LineResult object."""
-        from promptflow._sdk._submitter.test_submitter import TestSubmitterViaProxy
+        from promptflow._sdk._submitter import TestSubmitter
         from promptflow._sdk.operations._flow_context_resolver import FlowContextResolver
 
         if self.language == FlowLanguage.CSharp:
-            with TestSubmitterViaProxy(flow=self, flow_context=self.context).init() as submitter:
-                result = submitter.exec_with_inputs(
+            with TestSubmitter(flow=self, flow_context=self.context).init() as submitter:
```

Comment:
I think for --ui secenario, when exit thie function, executor will be destroied?  I think this is not expected since we need get generator results in streamlit main.py

Reply:
Yes, there will be a further refactor in the PR to enable streaming. Want to make current PR simpler for review